src/main/java/com/teragrep/lsh_01/pool/ManagedRelpConnection.java

```diff
@@ -51,38 +51,33 @@ void connect() {
                                 e.getMessage()
                         );
             }
-            if (connected) {
-                notConnected = false;
+
+            try {
+                Thread.sleep(relpConnection.relpConfig().relpReconnectInterval);
             }
-            else {
-                try {
-                    Thread.sleep(relpConnection.relpConfig().relpReconnectInterval);
-                }
-                catch (InterruptedException e) {
-                    LOGGER.error("Reconnect timer interrupted, reconnecting now");
-                }
+            catch (InterruptedException e) {
+                LOGGER.error("Reconnect timer interrupted, reconnecting now");
             }
         }
     }
 
     private void tearDown() {
-        relpConnection.tearDown();
-    }
-
-    private void disconnect() {
-        boolean disconnected = false;
-        try {
-            disconnected = relpConnection.disconnect();
-        }
-        catch (IllegalStateException | IOException | TimeoutException e) {
-            LOGGER.error("Forcefully closing connection due to exception <{}>", e.getMessage());
-        }
-        finally {
-            this.tearDown();
+        /*
+         TODO remove: wouldn't need a check but there is a bug in RLP-01 tearDown()
+         see https://github.com/teragrep/rlp_01/issues/63 for further info
+         */
+        if (connected) {
+            relpConnection.tearDown();
```

Comment:
tearDown must take place anyhow, it is the only way to cleanup interrupted network connection.

Reply:
Is this correct now with hasConnected? Now the tearDown is only executed if the connection has been made at least once. Now it does the tearDown even if it has been disconnected after an initial connection, unlike before.